apps/prairielearn/src/pages/instructorRequestCourse/instructorRequestCourse.html.ts

```diff
@@ -0,0 +1,247 @@
+import { html } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+import { type CourseRequestRow } from './instructorRequestCourse';
+
+export const RequestCourse = ({
+  course_requests,
+  resLocals,
+}: {
+  course_requests: CourseRequestRow[];
+  resLocals: Record<string, any>;
+}) => {
+  return html`
+    <!doctype html>
+    <html lang="en">
+      <head>
+        ${renderEjs(__filename, "<%- include('../partials/head')%>", {
+          ...resLocals,
+        })}
+        <script>
+          $(function () {
+            $('input[name=cr-role]').change(function () {
+              var role = this.value;
+              $('.question-form button').prop('disabled', role != 'instructor');
+              $('.role-comment').hide();
+              $('.role-comment-' + role).show();
+            });
+          });
+        </script>
+      </head>
+      <body>
+        ${renderEjs(__filename, "<%- include('../partials/navbar')%>", {
+          ...resLocals,
+          navPage: 'request_course',
+        })}
+        <main id="content" class="container">
+          ${course_requests.length > 0
```

Comment:
Put this block into a separate function (at the end of the file) and just include it here. That is, here we'd have something like `${CourseRequestsCard({ rows, csrfToken: resLocals.__csrfToken })}`. Note that we do not pass the entire `resLocals` into these helper functions.

The helper function itself would look like:
```
function CourseRequestsCard({rows, csrfToken}: {rows: CourseRequestRow[], csrfToken: string}): HtmlSafeString {
}
```

Reply:
Changed in 9664c3b5